src/bootupd.rs

```diff
@@ -244,7 +244,7 @@ pub(crate) fn update(name: &str, rootcxt: &RootContext) -> Result<ComponentUpdat
         .context("Failed to update state")?;
 
     let newinst = component
-        .run_update(&state_guard.sysroot, &inst)
+        .run_update(rootcxt, &inst)
```

Comment:
I think this is fine, though it probably makes sense to fold the SysrootGuard into being a lock on RootContext or so.

Reply:
Not sure I catch your meaning, current `StateLockGuard` locks `sysroot` Dir, do you mean to add `RootContext` to `StateLockGuard` ?

Reply:
It's all good as is!